const catchAsyncErrors = require("../middlewares/catchAsyncErrors.js");
const Brand = require("../models/brandModel.js");
const validateMongoDbId = require("../utils/validateMongodbId.js");

exports.createBrand = catchAsyncErrors(async (req, res) => {
  try {
    const newBrand = await Brand.create(req.body);
    res.json(newBrand);
  } catch (error) {
    throw new Error(error);
  }
});
exports.updateBrand = catchAsyncErrors(async (req, res) => {
  const { id } = req.params;
  validateMongoDbId(id);
  try {
    const updatedBrand = await Brand.findByIdAndUpdate(id, req.body, {
      new: true,
    });
    res.json(updatedBrand);
  } catch (error) {
    throw new Error(error);
  }
});
exports.deleteBrand = catchAsyncErrors(async (req, res) => {
  const { id } = req.params;
  validateMongoDbId(id);
  try {
    const deletedBrand = await Brand.findByIdAndDelete(id);
    res.json({
      message: "Product deleted successfully",
    })
  } catch (error) {
    throw new Error(error);
  }
});
exports.getBrand = catchAsyncErrors(async (req, res) => {
  const { id } = req.params;
  validateMongoDbId(id);
  try {
    const getaBrand = await Brand.findById(id);
    res.json(getaBrand);
  } catch (error) {
    throw new Error(error);
  }
});
exports.getallBrand = catchAsyncErrors(async (req, res) => {
  try {
    const getallBrand = await Brand.find();
    res.json(getallBrand);
  } catch (error) {
    throw new Error(error);
  }
});
